client/components/gravatar/index.jsx

```diff
@@ -1,73 +1,103 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { Component, PropTypes } from 'react';
 import url from 'url';
 import qs from 'querystring';
+import { connect } from 'react-redux';
+import { get } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import safeImageURL from 'lib/safe-image-url';
+import {
+	getUserTempGravatar
+} from 'state/current-user/gravatar-status/selectors';
 
-module.exports = React.createClass( {
-	displayName: 'Gravatar',
+export class Gravatar extends Component {
+	constructor() {
+		super( ...arguments );
+		this.state = {
+			failedToLoad: false
+		};
+	}
 
-	propTypes: {
-		user: React.PropTypes.object,
-		size: React.PropTypes.number,
-		imgSize: React.PropTypes.number
-	},
+	static propTypes = {
+		user: PropTypes.object,
+		size: PropTypes.number,
+		imgSize: PropTypes.number,
+		// connected props:
+		tempImage: PropTypes.oneOfType( [
+			PropTypes.string, // the temp image base64 string if it exists
+			PropTypes.bool // or false if the temp image does not exist
+		] ),
+	};
 
-	getDefaultProps() {
+	static defaultProps = {
 		// The REST-API returns s=96 by default, so that is most likely to be cached
-		return {
-			imgSize: 96,
-			size: 32
-		};
-	},
-
-	getInitialState() {
-		return {
-			failedToLoad: false
-		};
-	},
+		imgSize: 96,
+		size: 32
+	};
 
 	getResizedImageURL( imageURL ) {
+		const { imgSize } = this.props;
 		imageURL = imageURL || 'https://www.gravatar.com/avatar/0';
 		const parsedURL = url.parse( imageURL );
 		const query = qs.parse( parsedURL.query );
 
 		if ( /^([-a-zA-Z0-9_]+\.)*(gravatar.com)$/.test( parsedURL.hostname ) ) {
-			query.s = this.props.imgSize;
+			query.s = imgSize;
 			query.d = 'mm';
 		} else {
 			// assume photon
-			query.resize = this.props.imgSize + ',' + this.props.imgSize;
+			query.resize = imgSize + ',' + imgSize;
 		}
 
 		parsedURL.search = qs.stringify( query );
 		return url.format( parsedURL );
-	},
+	}
 
-	onError() {
+	onError = () => {
 		this.setState( { failedToLoad: true } );
-	},
+	};
 
 	render() {
-		const size = this.props.size;
+		const {
+			alt,
+			size,
+			tempImage,
+			user,
+		} = this.props;
 
-		if ( ! this.props.user ) {
+		if ( ! user ) {
 			return <span className="gravatar is-placeholder" style={ { width: size, height: size } } />;
 		} else if ( this.state.failedToLoad ) {
 			return <span className="gravatar is-missing" />;
 		}
```

Comment:
I see that this was here already, but we do try and avoid these `else if` constructions. it would be a nice opportunity to update

```js
if ( ! user ) {
	return …
}

if ( this.state.failedToLoad ) {
	return …
}
```